api/src/socketio.ts

```diff
@@ -103,54 +79,41 @@ export class Socketio {
     @SubscribeMessage("data")
     async onClientData(@MessageBody() data: any, @ConnectedSocket() socket: any) {
         // TODO: Get userId from JWT or determine if public user and link to configurable "public" user doc
-        socket.data.user = "editor-private";
-        socket.data.groups = ["group-private-editors"];
-
-        // Validate change request document
-        const changeReq = plainToInstance(ChangeReqDto, data);
-        const changeReqValidation = await validate(changeReq);
-        if (changeReqValidation.length > 0) {
-            let message = "Change request validation failed for the following constraints:\n";
-            changeReqValidation.forEach((c) => {
-                message += Object.values(c.constraints).join("\n") + "\n";
-            });
+        socket.data.user = "super-admin";
+        socket.data.groups = ["group-super-admins"];
+
+        // Validate received data
+        const message = await validateChangeReq(data);
+        if (message) {
             this.emitAck(socket, AckStatus.Rejected, data.reqId, message);
             return;
         }
 
-        // Check included document existance and type validity
-        if (!changeReq.doc.type || !Object.values(DocType).includes(changeReq.doc.type)) {
-            this.emitAck(
-                socket,
-                AckStatus.Rejected,
-                data.reqId,
-                `Submitted "${changeReq.doc.type}" document validation failed:\nInvalid document type`,
-            );
+        // Reject non user-editable types
+        // TODO: add test
```

Comment:
Just to check if this is still relevant.

Reply:
It is still relevant, but we need to think how we are going to do Socket.io specific testing when authentication is implemented.

An option would here would be to move the check to the permission system, where it will be easier to test.

Reply:
Document type checking moved to PermissionSystem.ts